user-tools/README-CVText.md

```diff
@@ -0,0 +1,114 @@
+  - The `CVText` type can be used in Bot Segments for both `endCriteria` and/or `botAction`.  This type looks for the presence of the supplied text in the current frame of the game.  This type communicates with our remote AIService instance to perform the evaluation.  When a result is found, the AIService provides the bounding rect information to our SDK to confirm the existence and location of the supplied text in the current frame.
+
+**TODO:** [Loom Video - Feel free to ignore this as I will be making these from my own demos]
+
+## Configuration
+
+### Key JSON fields
+
+#### `endCriteria`
+- `type` - `CVText` - tells the SDK to evaluate this type of criteria using the supplied data
+- `transient` - `transient`=`true` means that this text can match at any time during the evaluation of this bot segment and that passing result will persist even if it takes multiple more frames before other criteria in this segment are matched.  `transient`=`false` means that his criteria and other non-transient criteria must all be true at the same time (any transient criteria must also have matched already).  For `CVText` evaluation, it is not recommended to mark `transient`=`false`.  **Transient should almost always be `true` for `CVText`**.
+- `data` - The data json object that defines how to evaluate this `CVText` criteria.
+  - `text` - The text to find.  Note that this algorithm will treat this as a set of words to find, NOT a sentence.  Thus if you have 'Create New Profile' on a single line on one button, but have 'Create' 'New' 'Profile' each on a separate line in different screen area, the algorithm has to choose which one to select.  The current method is to select the smallest bounding area containing all required words that is also `withinRect` (if specified).
+  - `textMatchingRule` - One of `Matches` or `Contains`
+    - `Matches` - Each word in the provided `text` must be found as a whole word in the results.  This is a very exact matching rule and can sometimes lead to inconsistent results in frames with poor in game lighting on the text or low text contrast.
+    - `Contains` - Each word in the provided `text` must be found as a part of a word in the results.  This is a looser matching rule and if often used instead of `Matches` for more stable results.  For example "Time" would match to "Timer", "Time", "Times", "Timed", "Time", etc in the frame.  This may not seem ideal but, in most game situations, the contrast or text layout isn't always clearly identifiable and extra or incorrect letters may consistently be found for the text you are looking for in a specific frame of the game.  When you encounter those situations, you can adjust your game to give better text clarity/contrast to your users and/or you can utilize `Contains` instead of `Matches`.
```

Comment:
```suggestion
    - `Contains` - Each word in the provided `text` must be found as a part of a word in the results.  This is a looser matching rule and is often used instead of `Matches` for more stable results.  For example "Time" would match to "Timer", "Time", "Times", "Timed", "Time", etc in the frame.  This may not seem ideal but, in most game situations, the contrast or text layout isn't always clearly identifiable and extra or incorrect letters may consistently be found for the text you are looking for in a specific frame of the game.  When you encounter those situations, you can adjust your game to give better text clarity/contrast to your users and/or you can utilize `Contains` instead of `Matches`.
```